pkg/datapath/loader/netlink.go

```diff
@@ -98,14 +99,26 @@ func replaceDatapath(ctx context.Context, ifName, objPath, progSec, progDirectio
 		}
 	}
 
+	// If the iproute2 call below is successful, any 'pending' map pins will be removed.
+	// If not, any pending maps will be re-pinned back to their initial paths.
 	cmd := exec.CommandContext(ctx, loaderProg, args...).WithFilters(libbpfFixupMsg)
 	if _, err := cmd.CombinedOutput(log, true); err != nil {
 		// Program/object replacement unsuccessful, revert bpffs migration.
-		revert = true
-		return fmt.Errorf("Failed to load prog with %s: %w", loaderProg, err)
+		if err := bpf.FinalizeBPFFSMigration(bpf.MapPrefixPath(), objPath, true); err != nil {
+			return nil, fmt.Errorf("Failed to revert bpffs map migration: %w", err)
+		}
+		return nil, fmt.Errorf("Failed to load prog with %s: %w", loaderProg, err)
 	}
 
-	return nil
+	finalize := func() {
+		l := log.WithField("device", ifName).WithField("objPath", objPath)
```

Comment:
FYI there's also `log.WithFields(logrus.Fields{ ... })` to specify multiple fields at once with one call. This is a more common style elsewhere in the codebase, but also this doesn't make any meaningful difference so don't worry about changing this one.